Content.Shared/Movement/Components/SpriteMovementComponent.cs

```diff
@@ -0,0 +1,22 @@
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Movement.Components;
+
+/// <summary>
+/// Updates a sprite layer based on whether an entity is moving via input or not.
+/// </summary>
+[RegisterComponent, NetworkedComponent]
+public sealed partial class SpriteMovementComponent : Component
+{
+    /// <summary>
+    /// Layer and sprite state to use when moving.
+    /// </summary>
+    [DataField]
+    public Dictionary<string, string> MovementLayers = new();
+
+    /// <summary>
+    /// Layer and sprite state to use when not moving.
+    /// </summary>
+    [DataField]
+    public Dictionary<string, string> NoMovementLayers = new();
```

Comment:
`PrototypeLayerData` for the value instead of string state so color/visibility/etc can be changed?